extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -223,48 +221,91 @@ export function getPlacementsFromConfigObj(win, configObj) {
   }
-  const anchorElement = getAnchorElement(win, anchor);
-  if (!anchorElement) {
+  const anchorElements = getAnchorElements(win, anchor);
+  if (!anchorElements.length) {
     dev().warn(TAG, 'No anchor element found');
-    return null;
-  }
-  if ((placementObj['pos'] == Position.BEFORE ||
-       placementObj['pos'] == Position.AFTER) &&
-      !anchorElement.parentNode) {
-    dev().warn(TAG, 'Parentless anchor with BEFORE/AFTER position.');
-    return null;
+    return [];
   }
-  return new Placement(win, resourcesForDoc(anchorElement), anchorElement,
-      placementObj['pos'], injector);
+  const placements = [];
+  anchorElements.forEach(anchorElement => {
+    if ((placementObj['pos'] == Position.BEFORE ||
+         placementObj['pos'] == Position.AFTER) &&
+        !anchorElement.parentNode) {
+      dev().warn(TAG, 'Parentless anchor with BEFORE/AFTER position.');
+      return;
+    }
+    placements.push(new Placement(win, resourcesForDoc(anchorElement),
+        anchorElement, placementObj['pos'], injector));
+  });
+  return placements;
 }
 
 /**
+ * Looks up the element(s) addresses by the anchorObj.
+ *
+ * For a standard anchorObj this will return a single item array, or an empty
+ * array is no element on the page was addressed by the anchorObj.
+ *
+ * For a anchorObj containing a sub-anchor, sub-elements of the main selected
+ * element will be looked up and all matching elements returned. An example
+ * of how the sub-anchor feature might be used is looking up all <P> elements
+ * contained within the main content section of a page (where the main anchorObj
+ * selector identifies the main content section and the sub-anchor identifies a
+ * variable number of <P> elements).
+ *
  * @param {!Window} win
  * @param {!Object} anchorObj
- * @return {?Element}
+ * @return {!Array<!Element>}
  */
-function getAnchorElement(win, anchorObj) {
+function getAnchorElements(win, anchorObj) {
   const selector = anchorObj['selector'];
   if (!selector) {
     dev().warn(TAG, 'No selector in anchor');
-    return null;
+    return [];
   }
   const index = anchorObj['index'] || 0;
-  if (index == 0) {
-    return win.document.querySelector(selector);
+  const mainEl = index == 0 ?
+      win.document.querySelector(selector) :
+      win.document.querySelectorAll(selector)[index];
+  if (!mainEl) {
+    return [];
+  }
+  if (anchorObj['sub']) {
+    return getSubAnchorElements(mainEl, anchorObj['sub']);
+  }
+  return [mainEl];
+}
+
+/**
+ * @param {!Element} ancestorElement
+ * @param {!Object} subAnchor
+ * @return {!Array<!Element>}
+ */
+function getSubAnchorElements(ancestorElement, subAnchor) {
+  const selector = subAnchor['selector'];
+  if (!selector) {
+    dev().warn(TAG, 'No selector in sub-anchor');
+    return [];
   }
-  return win.document.querySelectorAll(selector)[index] || null;
+  const minChars = subAnchor['min_c'] || 0;
+  const subElements = [].slice.call(ancestorElement.querySelectorAll(selector));
```

Comment:
Generating a unique selector doesn't sound trivial? I feel that we're maybe proposing to add quite a bit of complexity for not a lot of functionality. I think just explaining that the sub-selector will be evaluated within the context of the DOM is fine, as long as anyone generating a config is aware of this.

I've put the code back to what it was initially but with a warning. If it's OK with you, I'd rather just leave it like that.